automation/check-patch.e2e-linux-bridge-functests.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+set -xeu
+
+# This script should be able to execute linux-bridge
+# functional tests against Kubernetes cluster with
+# CNAO built with latest changes, on any
+# environment with basic dependencies listed in
+# check-patch.packages installed and docker running.
+#
+# yum -y install automation/check-patch.packages
+# automation/check-patch.e2e-linux-bridge-functests.sh
+
+teardown() {
+    cp $(find . -name "*junit*.xml") $ARTIFACTS || true
+    cluster/kubectl.sh logs -n cluster-network-addons -l app=cni-linux-bridge-plugin > $ARTIFACTS/linux-bridge.log || true
+    cluster/kubectl.sh get nodes -o yaml > $ARTIFACTS/nodes.yaml || true
+    rm -rf "${TMP_COMPONENT_PATH}"
+    cd ${TMP_PROJECT_PATH}
+    make cluster-down
+}
+
+build_binaries() {
+	./build_linux.sh
+}
+
+collect_binaries_shasum_from_repo() {
+	sha_file=$1
+	build_binaries
+  sha256sum bin/* >${sha_file}
+}
+
+collect_binaries_shasum_from_node() {
+	sha_file=$1
+	${TMP_PROJECT_PATH}/cluster/ssh.sh node01 -- sha256sum /opt/cni/bin/* >${sha_file}
+}
+
+extract_sha_from_shasum_file() {
+	file_name_in_sha=$1
+	sha_file=$2
+	grep ${file_name_in_sha1} ${sha_file1} | awk '{print $1}'
+}
+
+compare_sha_files() {
```

Comment:
I think this should be named assert_sha_files_eq, or something similar. Otherwise we "compare" but we do nothing with the result.

Reply:
Done

Reply:
This function API is a bit strange.
I can think of two options:
1. Pass in the two sha-sets and a list of keys to filter the content of the sets before the check.
Then, instead of `extract_sha_from_shasum_file`, change it a bit to `filter_shas_from_file`, providing the file and the list.
This will allow calling it once and not twice, and have a more reasonable input.
2. Just take lines 50-51 out, and call this function with their result directly.